staging/src/k8s.io/apiserver/pkg/server/options/serving.go

```diff
@@ -165,9 +165,16 @@ func (s *SecureServingOptions) applyServingInfoTo(c *server.Config) error {
 		return nil
 	}
 
+	ipMax := c.MaxRequestsInFlight / 2
+	if ipMax > c.MaxMutatingRequestsInFlight/2 {
+		ipMax = c.MaxMutatingRequestsInFlight / 2
+	}
+	max := c.MaxRequestsInFlight + c.MaxMutatingRequestsInFlight
```

Comment:
We've been trying to limit the need to expose additional config parameters. For small masters (1-4 CPU on master and a limited number of nodes) these values seem to be correct and provide much better resilience. However I think with differing sizes and deployment types, the sort of config option you suggested is probably the best choice.